### PR TITLE
PR: Convert LSPManager into a proper SpyderPlugin

### PR DESCRIPTION
**Pull request by andfoy**
## Description of Changes
This PR refactors the previously-known ``LSPManager`` into ``LanguageServerPlugin``, using the ``SpyderPlugin`` API introduced in #9226

* [x] Wrote at least one-line docstrings (for any new functions)
* [ ] Added unit test(s) covering the changes (if testable)

* [ ] Included a screenshot or animation (if affecting the UI, see [Licecap](https://www.cockos.com/licecap/))

### Issue(s) Resolved

Fixes #

### Affirmation

By submitting this Pull Request or typing my (user)name below,
I affirm the [Developer Certificate of Origin](https://developercertificate.org)
with respect to all commits and content included in this PR,
and understand I am releasing the same under Spyder's MIT (Expat) license.

I certify the above statement is true and correct:
@andfoy

### CODE REVIEW

File: spyder/api/plugins.py [hunk 1]
```diff
@@ -154,6 +154,10 @@ class SpyderPlugin(BasePlugin):
     # Status: Optional
     CONFIGWIDGET_CLASS = None
 
+    # The plugin does not have a UI component
+    # Status: Optional
+    NO_WIDGET = True
```
Comment (ccordoba12) [inline spyder/api/plugins.py]: There's no need of this. You need to check if the plugin has a `dockwidget` attribute. Only SpyderPluginWidget's have one.

File: spyder/api/plugins.py [hunk 2]
```diff
@@ -357,6 +361,8 @@ class SpyderPluginWidget(SpyderPlugin, BasePluginWidget):
     # Status: Optional
     shortcut = None
 
+    NO_WIDGET = False
```
Comment (ccordoba12) [inline spyder/api/plugins.py]: Please remove this too.

**Comment by ccordoba12** [comment]
Also, please rebase or merge with latest master to get the fixes to our tests.

**Comment by andfoy** [comment]
@ccordoba12 Is this one ready?

**Comment by ccordoba12** [comment]
Yes, but I don't want to include it as part of beta3. Since you need to do more work on top of this, please create a branch from here and then you'll rebase when this one is merged (in a day or two). Thanks!

**Pull request merged**